pkg/build/build.go

```diff
@@ -127,6 +126,19 @@ func New(opts ...Option) (*Context, error) {
 		}
 	}
 
+	// If no workspace directory is explicitly requested, create a
+	// temporary directory for it.  Otherwise, ensure we are in a
+	// subdir for this specific build context.
+	if ctx.WorkspaceDir != "" {
+		ctx.WorkspaceDir = filepath.Join(ctx.WorkspaceDir, ctx.Arch.ToAPK())
+	} else {
+		tmpdir, err := os.MkdirTemp("", "melange-workspace-*")
```

Comment:
should we clean this temp directory at the end of the process? or it is ok to keep it?

Reply:
It gets cleaned up later on.  If we do a `defer` here, it will get cleaned up too early.

Reply:
yep, not here.

ah if that cleans up later that is cool